app/uk/gov/hmrc/gform/sharedmodel/TaxPeriods.scala

```diff
@@ -61,3 +62,41 @@ case class TaxResponse(id: HmrcTaxPeriod, obligation: Obligation)
 object TaxResponse {
   implicit val format: OFormat[TaxResponse] = Json.format[TaxResponse]
 }
+
+case class TaxPeriodIdentifier(idType: IdType, idNumber: IdNumber, regimeType: RegimeType)
+
+object TaxPeriodIdentifier {
+  implicit val format: OFormat[TaxPeriodIdentifier] = Json.format[TaxPeriodIdentifier]
+}
+
+case class AllInfo(
+  idType: IdType,
+  idNumber: TextExpression,
+  regimeType: RegimeType,
+  inboundCorrespondenceFromDate: Date,
+  inboundCorrespondenceToDate: Date,
+  periodKey: String)
+
+object AllInfo {
+  implicit val format: OFormat[AllInfo] = derived.oformat
+}
+
+case class ListAllInfo(listAllInfo: List[AllInfo])
+
+object ListAllInfo {
+  implicit val format: OFormat[ListAllInfo] = Json.format[ListAllInfo]
+
+  implicit val optionFormat: OFormat[Option[ListAllInfo]] = new OFormat[Option[ListAllInfo]] {
```

Comment:
Why do we need custom `OFormat`?

Reply:
The default OFormat doesn't work with Options as far as I can tell

Reply:
> doesn't work
What you mean by that? It doesn't compile, it works incorrectly or something else? We use `Option` all over the place...